### DISCUSSION TITLE
Is there way to order produced queries in ORM mode?

### DISCUSSION THREAD

**Discussion by artofhuman** [sqlalchemy/sqlalchemy#8900] [Usage Questions]
There is no way to order produced queries (ORM mode). It causes SQL error in case if table have unique constraint.

For example, the table have unique index `"contractor_idx" UNIQUE, btree (inn, is_main)`

``` python
# just fill db
_c1 = f.ContractorFactory(inn=INN, is_main=None)
_c2 = f.ContractorFactory(inn=INN, is_main=1)

c1 = db.get(Contractor, _c1.id)
c2 = db.get(Contractor, _c2.id)

c1.is_main = 1
c2.is_main = None

db.commit()

```
In this case, the first query will be 
```
UPDATE contractor SET is_main=1 WHERE contractor.id = 1
```
which cause the error
```
duplicate key value violates unique constraint "contractor_idx"
```

What I've tried
1. it does not work
```
db.add(c2)
db.add(c1)
db.commit()
```
2. It does not work, and i think it's a bug
```
db.flush([c2, c1])
```

3. It works, but it's looks like not clear solution.
```
db.flush([c2])
db.flush([c1])
```

db - postgresql
sqlalchemy - 1.4.44

**Comment by CaselIT** [comment]
Hi,

Sorry this was forgot. No at the moment there is no way of ordering the operations that are done by the orm.
You will need to use flush to force an ordering.

This will also work
```py
# just fill db
_c1 = f.ContractorFactory(inn=INN, is_main=None)
_c2 = f.ContractorFactory(inn=INN, is_main=1)

c1 = db.get(Contractor, _c1.id)
c2 = db.get(Contractor, _c2.id)

c2.is_main = None
db.flush()
c1.is_main = 1

db.commit()
```
